docs/src/manual/constraints.md

```diff
@@ -55,4 +56,24 @@ apply!(K, f, ch)
 a = K\f
 apply!(a, ch) # enforces affine constraints
 
-```
\ No newline at end of file
+```
+
+### Solving nonlinear problems
+It is important to check the residual **after** applying boundary conditions when 
+solving nonlinear problems with affine constraints. 
+`apply_zero!(K, r, ch)` modifies the residual entries for dofs that are involved 
+in constraints to account for constraint forces. 
+The following pseudo-code shows a typical pattern for solving a non-linear problem with Newton's method:
+```julia
+a = initial_guess(...)  # Make any initial guess for a here, e.g. `a=zeros(ndofs(dh))`
+Δa = similar(a)         # Preallocate Δa
+apply!(a, ch)           # Make the guess fulfill all constraints in `ch`
+for iter in 1:maxiter
+    doassemble!(K, r, ...)  # Assemble the residual, r, and stiffness, K=∂r/∂a.
+    apply_zero!(K, r, ch)   # Modify `K` and `r` to account for the constraints. 
+    check_convergence(r, ...) && break # Only check convergence after `apply_zero!(K, r, ch)`
+    Δa .= .- (K \ r)        # Calculate update
```

Comment:
```suggestion
    Δa .-= K \ r            # Calculate update
```

Reply:
I misunderstood the code I think. It doesn't seem necessary to shuffle the data to a preallocated Δa here, just call `Δa = K \ r` and then `a .-= Δa`?